modules/core/include/opencv2/core/ocl.hpp

```diff
@@ -697,7 +697,8 @@ class CV_EXPORTS PlatformInfo
     Impl* p;
 };
 
-CV_EXPORTS const char* convertTypeStr(int sdepth, int ddepth, int cn, char* buf);
+CV_EXPORTS CV_DEPRECATED const char* convertTypeStr(int sdepth, int ddepth, int cn, char* buf);
+CV_EXPORTS String convertTypeStr(int sdepth, int ddepth, int cn);
```

Comment:
Could we add `buf_size` parameter instead?

Reply:
Instead of returning `String`?  i.e. you would prefer:

`CV_EXPORTS const char* convertTypeStr(int sdepth, int ddepth, int cn, char* buf, size_t buf_size);`

Reply:
Yes, lets avoid calls for heap memory allocation.